torch/csrc/autograd/engine.cpp

```diff
@@ -186,19 +186,19 @@ struct GraphTask {
   // Safe to read owner_ and reentrant_depth_ without synchronizaton
   int owner_;
   // The number of parent graph tasks for this graph task
-  int reentrant_depth_;
+  const int reentrant_depth_;
 
   bool can_checkpoint() {
     return exec_info_.empty();
   }
 
-  GraphTask(bool keep_graph, bool grad_mode)
+  GraphTask(bool keep_graph, bool grad_mode, int reentrant_depth = 0)
```

Comment:
Are you sure you want this to be an optional argument? It might hide some bugs where you need to pass in reentrant depth but you forget because it defaults to 0.